app/src/main/java/com/android/voyageur/ui/trip/AddActivity.kt

```diff
@@ -84,12 +79,27 @@ fun AddActivityScreen(tripsViewModel: TripsViewModel, navigationActions: Navigat
 
     val today = normalizeToMidnight(Date())
     val dateNormalized = activityDate?.let { normalizeToMidnight(Date(it)) } ?: Date(0)
+    val selectedTrip = tripsViewModel.selectedTrip.value!!
+
+    if (activityDate != null &&
+        (dateNormalized.after(normalizeToMidnight(selectedTrip.endDate.toDate())) ||
+            dateNormalized.before(normalizeToMidnight(selectedTrip.startDate.toDate())))) {
+      Toast.makeText(
+              context, "The activity date is not within the trip's dates", Toast.LENGTH_SHORT)
```

Comment:
nit: What do you think of the Toast message being "Please select a date within the trip's scheduled dates" instead? Up to you

Reply:
Indeed, I think your toast suggestion is more user friendly, I'll change it.